backend/src/data/migrations/20220822100917_add_fullname_to_user_details.ts

```diff
@@ -0,0 +1,51 @@
+import { Knex } from 'knex';
+
+enum ColumnName {
+  FULL_NAME = 'full_name',
+  FIRST_NAME = 'first_name',
+  LAST_NAME = 'last_name',
+}
+
+enum TableName {
+  USERS = 'users',
+  USER_DETAILS = 'user_details',
+}
+
+async function up(knex: Knex): Promise<void> {
+  await knex.schema.table(TableName.USER_DETAILS, (table) => {
+    table.dropColumn(ColumnName.FIRST_NAME);
+    table.dropColumn(ColumnName.LAST_NAME);
+    table.string(ColumnName.FULL_NAME).notNullable();
+  });
+
+  await knex.raw(
+    'INSERT INTO user_details (user_id, full_name) ' +
+      '(SELECT id AS user_id, full_name ' +
+      'FROM users)',
+  );
+
+  await knex.schema.table(TableName.USERS, (table) => {
+    table.dropColumn(ColumnName.FULL_NAME);
+  });
+}
+
+async function down(knex: Knex): Promise<void> {
+  await knex.schema.table(TableName.USER_DETAILS, (table) => {
+    table.string(ColumnName.FIRST_NAME).notNullable();
+    table.string(ColumnName.LAST_NAME).notNullable();
+  });
```

Comment:
won't this fail if there are records in user_details?

Reply:
As it should be a full fallback to existed table structure it could be accomplished by: at first just turningin back `first_name`, `last_name` cols, restoring it's values from `full_name` col, then adding `not nullable` constraints
@PavelSukhinin what do you think?

Reply:
@andanfe sounds like a plan 👍